.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+default_language_version:
+  python: python3
+
+exclude: ^$|instrument
+
+repos:
+
+  # Run fast code improvement/checks before running PR specific helpers.
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: master
+    hooks:
+      - id: check-xml
+      - id: check-yaml
+
+  - repo: https://github.com/mantidproject/pre-commit-hooks
+    rev: main
+    hooks:
+      - id: clang-format
+        exclude: Testing/Tools/cxxtest|tools
+
+  - repo: https://github.com/google/yapf
+    rev: main
+    hooks:
+      - id: yapf
+
+  # Flake8 isn't automatically parallel on Windows due to lack of multiprocessing library support. So default to 16 jobs
+  # to support 8 hyper threaded cores, In practice the drop on other OS' should be negligible
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.7.9
+    hooks:
+      - id: flake8
+        # Have to manually copy this from the .flake8 as flake8 will just ignore the config if the file is passed
+        exclude: build|buildconfig|docs|Framework/Algorithms/test|Framework/DataObjects|Framework/MPIAlgorithms|Framework/PythonInterface/mantid|Framework/PythonInterface/test|images|installers|instrument|MantidPlot|qt/applications/workbench/workbench/app/resources.py|qt/paraview_ext|qt/python/mantidqt/resources.py|qt/python/mantidqt/dialogs/errorreports/resources_qt4.py|qt/python/mantidqt/dialogs/errorreports/resources_qt5.py|qt/scientific_interfaces|qt/widgets|Testing/PerformanceTests|Testing/SystemTests/lib|Testing/SystemTests/scripts|Testing/SystemTests/tests/framework/reference|Testing/Tools|scripts/ErrorReporter/ui_errorreport.py|scripts/Interface/ui|scripts/Interface/ui/dataprocessorinterface
+        args: ['--extend-ignore=E999,F821,E127', '-j16']
```

Comment:
Just because my editor will add the line if I go near this file, please add EOL to the last line.

Reply:
Done :+1: